apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -148,6 +148,58 @@ const DanceAiModal: React.FunctionComponent = () => {
     (state: {dance: DanceState}) => state.dance.aiOutput
   );
 
+  const getGeneratedEffect = useCallback((step: number) => {
+    if (step < BAD_GENERATED_RESULTS_COUNT) {
+      return generatedEffects.current.badEffects[step];
+    } else if (generatedEffects.current.goodEffect) {
+      return generatedEffects.current.goodEffect;
+    } else {
+      return undefined;
+    }
+  }, []);
+
+  const getScores = useCallback(
+    (useInputs: string[], step: number) => {
+      const effect = getGeneratedEffect(step);
+      if (effect) {
+        return getGeneratedEffectScores(useInputs, effect);
+      }
+
+      return [0, 0, 0];
+    },
+    [getGeneratedEffect]
+  );
+
+  // Calculates the minimum individual score
+  // (ie, a SINGLE emoji association with a foreground/background palette combination),
+  // and a maximum total score
+  // (ie, the sum of ALL selected emoji's associations with a foreground/background palette combination).
+  // Used to normalize and scale the data for easier differentiation between results by the user.
```

Comment:
Not blocking, but...

This `reduce` call feels excessively clever to me and like it's hiding the logic. I get that it's building a range and that javascript stupidly doesn't have one of those built in, but that initial definition is tough to read at first (at least for me). And then there's an Infinity, it's doing a `min` on the individual scores, and a comparison, and returning the final minimum value. The logic feels tough to parse out.

What about something like this instead?
```
const f = Math.min(                               // give Math.min a list of all the possible scores and find the single low point
  ...Array(BAD_GENERATED_RESULTS_COUNT + 1)       // array of the appropriate length
    .fill(undefined)                              // fill it with undefined so we can map over it
    .map((_, step) => getScores(useInputs, step)) // map the step to the getScores value
    .flat()                                       // flatten out the sub arrays into a single array, which is consumed by the Math.min
)
```